packages/web-components/fast-components/src/tabs/tabs.ts

```diff
@@ -0,0 +1,324 @@
+                        : `grid-row: ${index + 1};`
+                );
+                tab.addEventListener("click", this.handleTabClick);
+                tab.addEventListener("keydown", this.handleTabKeyDown);
+                tab.setAttribute("tabindex", this.activeTabIndex === index ? "0" : "-1");
+                if (this.activeTabIndex === index) {
+                    this.activeTab = tab;
+                }
+                !this.isHorizontal()
+                    ? tab.classList.add("vertical")
+                    : tab.classList.remove("vertical");
+            }
+        });
+    };
+
+    private getTabPanels = (): void => {
+        this.tabIDs = this.getTabIds();
+        this.tabPanelIDs = this.getTabPanelIds();
+        this.tabPanels.forEach((tabpanel: HTMLElement, index: number) => {
+            tabpanel.setAttribute(
+                "id",
+                this.tabPanelIDs[index] === undefined || this.tabPanelIDs[index] === null
+                    ? `panel-${index + 1}`
+                    : this.tabPanelIDs[index]
+            );
+            tabpanel.setAttribute(
+                "aria-labeledby",
+                this.tabIDs[index] === undefined || this.tabIDs[index] === null
+                    ? `tab-${index + 1}`
+                    : this.tabIDs[index]
+            );
+            this.activeTabIndex !== index
+                ? tabpanel.setAttribute("hidden", "")
+                : tabpanel.removeAttribute("hidden");
+        });
+    };
+
+    private getTabIds(): string[] {
+        return this.tabs.map((tab: HTMLElement) => {
+            return tab.getAttribute("id") as string;
+        });
+    }
+
+    private getTabPanelIds(): string[] {
+        return this.tabPanels.map((tabPanel: HTMLElement) => {
+            return tabPanel.getAttribute("id") as string;
+        });
+    }
+
+    private setTabs(): void {
+        this.activeId = this.tabIDs[this.activeTabIndex];
+        this.change();
+        this.getTabs();
+        this.handleActiveIndicatorPosition();
+        this.getTabPanels();
+        this.focusTab();
+        this.change();
+    }
+
+    private handleTabClick = (event): void => {
+        const selectedTab = event.srcElement as Element;
+        this.prevActiveTabIndex = this.activeTabIndex;
+        this.activeTabIndex = Array.from(this.tabs).indexOf(event.target);
+        if (selectedTab.nodeType === 1) {
+            this.setTabs();
+        }
+    };
+
+    private isHorizontal(): boolean {
+        return this.orientation === TabsOrientation.horizontal;
+    }
+
+    private handleTabKeyDown = (event: KeyboardEvent): void => {
+        const keyCode: number = event.keyCode;
+        const tabsLength: number = this.tabs.length;
+        if (this.isHorizontal()) {
+            switch (keyCode) {
+                case keyCodeArrowLeft:
+                    event.preventDefault();
+                    this.decrement(tabsLength);
```

Comment:
I'm not sure it's helpful to pass the tabLength to these functions - it's always the same and always retrieved from `this.tabs.length` - perhaps these functions could just access the property internally?

I could also see this being simplified by replacing `increment` and `decrement` with a single function that accepts a delta to adjust by:

```
this.adjust(-1)

//... 

this.adjust(1)
```